File: app/src/main/java/com/lee/leewanandroid/algorithm/java/tree/ITreeAction.java
package com.lee.leewanandroid.algorithm.java.tree;

import com.lee.leewanandroid.algorithm.java.tree.PrintStrategy.IPrintStrategy;
import com.lee.leewanandroid.algorithm.java.tree.node.Node;

/**
 * 树的一些基本操作，增、删、查、改、打印树
 */
public interface ITreeAction<K, V> {
    boolean insert(K key, V value);

    boolean remove(K key);

    Node<K, V> find(K key);

    boolean amend(K key, V value);

    void print(IPrintStrategy strategy);
}

File: app/src/main/java/com/lee/leewanandroid/algorithm/tree/ITreeAction.kt
package com.lee.leewanandroid.algorithm.tree

import com.lee.leewanandroid.algorithm.tree.node.Node

/**
 *
 * @Description:    ITreeAction 插入 删除 查找
 * @Author:         lihuayong
 * @CreateDate:     2019-06-17 18:23
 * @UpdateUser:
 * @UpdateDate:     2019-06-17 18:23
 * @UpdateRemark:
 * @Version:        1.0
 */
interface ITreeAction<T> {

    fun insert(value: T): Boolean

    fun remove(value: T): Boolean

    fun find(value: T): Node<T>?

    fun clear()
}